packages/abstractions/kiota_abstractions/serialization/parse_node.py

```diff
@@ -155,17 +156,17 @@ def get_enum_value(self) -> Enum:
         pass
 
     @abstractmethod
-    def get_object_value(self, factory: ParsableFactory) -> Parsable:
+    def get_object_value(self, factory: ParsableFactory[U]) -> U:
```

Comment:
should U be optional here?

Reply:
sorry, I didn't mean for the argument, but for the return type.

Reply:
I believe we're talking about the same thing. 
As `get_object_value` will call the `ParsableFactory` (which essentially calls `create_from_discriminator_value`). Since this never returns an optional, we won't get an optional returned as `get_object_value` simply populates the properties of the object returned by the factory and returns the modified object.

Reply:
It seems that we have disparities between Java and dotnet...

https://github.com/microsoft/kiota-dotnet/blob/3b1f96c2aa64a09792133838e4e0184038fa985f/src/abstractions/serialization/IParseNode.cs#L132

https://github.com/microsoft/kiota-java/blob/ce707fc2e11f3b2add49880365b47b87a5c2b62a/components/abstractions/src/main/java/com/microsoft/kiota/serialization/ParseNode.java#L160

But after thinking about it some more, I believe you're correct, GetObjectValue will only be called if the property is included and non null.

Good to resolve this one, I'll just create an issue in kiota dotnet.